server/src/main/kotlin/org/javacs/kt/overridemembers/OverrideMembers.kt

```diff
@@ -0,0 +1,282 @@
+package org.javacs.kt.overridemembers
+
+import org.eclipse.lsp4j.CodeAction
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.lsp4j.WorkspaceEdit
+import org.javacs.kt.CompiledFile
+import org.javacs.kt.util.toPath
+import org.javacs.kt.position.position
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtTypeArgumentList
+import org.jetbrains.kotlin.psi.KtSuperTypeListEntry
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.psi.KtSimpleNameExpression
+import org.jetbrains.kotlin.descriptors.Modality
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.DeclarationDescriptor
+import org.jetbrains.kotlin.descriptors.DescriptorVisibilities
+import org.jetbrains.kotlin.descriptors.ClassConstructorDescriptor
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.isInterface
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.js.resolve.diagnostics.findPsi
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.isAbstract
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.psi.psiUtil.unwrapNullability
+import org.jetbrains.kotlin.types.TypeProjection
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.types.typeUtil.asTypeProjection
+
+// TODO: see where this should ideally be placed
+private const val DEFAULT_TAB_SIZE = 4
+
+fun listOverridableMembers(file: CompiledFile, cursor: Int): List<CodeAction> {
+    val kotlinClass = file.parseAtPoint(cursor)
+
+    if (kotlinClass is KtClass) {
+        return createOverrideAlternatives(file, kotlinClass)
+    }
+
+    return emptyList()
+}
```

Comment:
We could in theory rewrite this to use a when-expression like above, but we need the `kotlinClass`-reference for the `createOverrideAlternatives`-call. That means the suggestion above won't compile due to unknown reference to `kotlinClass`. If we still need that variable, I don't think it improves much just using a when-expression instead of the if-version. Feel free to disagree 🙂